### PR TITLE
fix(DB): Brontus spawn and pathing behaviors corrected

### LINKED ISSUE #6539: The Barrens - Brontus Missing Patrol Pattern
#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [x] enUS
- [ ] other (specify)

##### FACTION

- [x] Alliance
- [x] Horde

##### CONTENT PHASE:

- [ ] Generic
- [x] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

##### CURRENT BEHAVIOUR:

Original report: https://github.com/chromiecraft/chromiecraft/issues/968

Brontus is stationary in his spawn point.
brontus

Tester: confirmed.
WoWScrnShot_062421_195548

##### EXPECTED BLIZZLIKE BEHAVIOUR:

He should be moving around. 
Going by wowhead comments, he should be patrolling around with a group of 3 other kodos.
https://classic.wowhead.com/npc=5827/brontus#comments
https://www.wowhead.com/npc=5827/brontus#comments

presumably one of these groups that patrols in barrens:
northern one
barrenskodos
GUID 15144 for Wooly Kodo

southern one:
kodos2

GUID 15139 (one of the kodos)

##### STEPS TO REPRODUCE THE PROBLEM:

1. Teleport to GUID 51815
2. Observe his stationary nature

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/34da0cda5195dede48cb0406b23248330d6249cb

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker),
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-Zonecheck](https://github.com/55Honey/acore_Zonecheck)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/99451439-the-barrens-brontus-missing-patrol-pattern?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by neifion-00000000**
closes #6539 and addresses all additional issues mentioned

## Changes Proposed:
-  Removes existing Brontus spawn
-  Creates 3 New Pooled spawns, containing Kodos that path (and Brontus in one of them)

## Issues Addressed:

- Closes #6539 and deals with a lot of the additional details found in research

## SOURCE:

sniff data, and a lot of research, the spawn percentages were guesses and might be too generous especially with how the pool spawning works which is also more predictable than would be desired

## Tests Performed:

- Checked Pooling functionality
- Checked respawn functionality (it's a bit odd but it doesn't break the pools ever)
- Checked pathing (it's a long path x.x)

## How to Test the Changes:

1. .go xyz -3787.7153 -1848.3755 93.918015
2. Have fun with the kodos for a while

## Known Issues and TODO List:

- Spawn percentage is prolly not perfect, but I spent more than 6 hours on this (killed the rare 3x), but its hard to tell for sure without spending hundreds of hours on it.
- My formations prolly aren't perfect.  I don't have a method for an exact formation.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Maelthyrr** [comment]
I got an issue with a pack. They are in each others. (One membe: guid 42811). I'm also unable to find brontus.

WoWScrnShot_061422_212315

File: data/sql/updates/pending_db_world/rev_1655187726583284000.sql [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+('45817', '393', 'Brontus Kodo Pack Combo 3 -- Brontus Pack Leader Mob'),
+('45818', '393', 'Brontus Kodo Pack Combo 3 -- Brontus Pack'),
+('45819', '393', 'Brontus Kodo Pack Combo 3 -- Brontus Pack'),
+('45820', '393', 'Brontus Kodo Pack Combo 3 -- Brontus Pack');
+
+DELETE FROM `pool_pool` WHERE pool_id IN (391, 392, 393);
+INSERT INTO `pool_pool` (`pool_id`, `mother_pool`, `chance`, `description`) VALUES 
+('391', '394', '55', 'Brontus Kodo Pack Combo 1 -- 4 Barrens Kodo Pack'),
+('392', '394', '30', 'Brontus Kodo Pack Combo 2 -- Mixed Kodo Pack'),
+('393', '394', '15', 'Brontus Kodo Pack Combo 3 -- Brontus Pack');
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1655187726583284000.sql:45]: remove ' from ints please :D

File: data/sql/updates/pending_db_world/rev_1655187726583284000.sql [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+('45813', '392', 'Brontus Kodo Pack Combo 2 -- Mixed Kodo Pack Leader Mob'),
+('45814', '392', 'Brontus Kodo Pack Combo 2 -- Mixed Kodo Pack'),
+('45815', '392', 'Brontus Kodo Pack Combo 2 -- Mixed Kodo Pack'),
+('45816', '392', 'Brontus Kodo Pack Combo 2 -- Mixed Kodo Pack'),
+('45817', '393', 'Brontus Kodo Pack Combo 3 -- Brontus Pack Leader Mob'),
+('45818', '393', 'Brontus Kodo Pack Combo 3 -- Brontus Pack'),
+('45819', '393', 'Brontus Kodo Pack Combo 3 -- Brontus Pack'),
+('45820', '393', 'Brontus Kodo Pack Combo 3 -- Brontus Pack');
+
+DELETE FROM `pool_pool` WHERE pool_id IN (391, 392, 393);
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1655187726583284000.sql:41]:
```suggestion
DELETE FROM `pool_pool` WHERE `pool_id` IN (391, 392, 393);
```

File: data/sql/updates/pending_db_world/rev_1655187726583284000.sql [hunk 3]
```diff
@@ -0,0 +1,185 @@
+--
+DELETE FROM `pool_template` WHERE entry IN (391, 392, 393); 
+INSERT INTO `pool_template` (`entry`, `max_limit`, `description`) VALUES ('391', '4', 'Brontus Kodo Pack Combo 1 -- 4 Barrens Kodo Pack');
+INSERT INTO `pool_template` (`entry`, `max_limit`, `description`) VALUES ('392', '4', 'Brontus Kodo Pack Combo 2 -- Mixed Kodo Pack');
+INSERT INTO `pool_template` (`entry`, `max_limit`, `description`) VALUES ('393', '4', 'Brontus Kodo Pack Combo 3 -- Brontus Pack');
+INSERT INTO `pool_template` (`entry`, `max_limit`, `description`) VALUES ('394', '1', 'Brontus Kodo Pack Pool of Pools');
+
+DELETE FROM `creature` WHERE guid IN (51815);
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1655187726583284000.sql:8]:
```suggestion
DELETE FROM `creature` WHERE `guid` IN (51815);
```

File: data/sql/updates/pending_db_world/rev_1655187726583284000.sql [hunk 4]
```diff
@@ -0,0 +1,185 @@
+--
+DELETE FROM `pool_template` WHERE entry IN (391, 392, 393);
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1655187726583284000.sql:2]:
```suggestion
DELETE FROM `pool_template` WHERE `entry` IN (391, 392, 393); 
```

File: data/sql/updates/pending_db_world/rev_1655187726583284000.sql [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+(@path,106,-3633.491,-1962.7506,91.96043,0,0,0,0,100,0),
+(@path,107,-3674.7666,-1972.0784,92.34276,0,0,0,0,100,0),
+(@path,108,-3718.9587,-1949.9241,94.24338,0,0,0,0,100,0),
+(@path,109,-3743.7683,-1944.3379,93.25019,0,0,0,0,100,0),
+(@path,110,-3765.317,-1912.2642,92.659584,0,0,0,0,100,0),
+(@path,111,-3780.6667,-1879.9966,94.51902,0,0,0,0,100,0),
+(@path,112,-3802.9849,-1847.3167,94.4449,0,0,0,0,100,0);
+-- 0x204CB0002003294000001C00001E1E9F .go xyz -3780.6667 -1879.9966 94.51902
+
+DELETE FROM `creature_formations` where leaderGUID in(45809, 45813, 45817);
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1655187726583284000.sql:172]:
```suggestion
DELETE FROM `creature_formations` WHERE `leaderGUID` IN (45809, 45813, 45817);
```

File: data/sql/updates/pending_db_world/rev_1655187726583284000.sql [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+SET @NPC := 45809;
+SET @path := @NPC * 10;
+DELETE FROM creature_addon WHERE guid=@NPC;
+DELETE FROM creature_addon WHERE guid=@NPC+4;
+DELETE FROM creature_addon WHERE guid=@NPC+8;
+INSERT INTO creature_addon (guid,path_id,mount,bytes1,bytes2,emote,visibilityDistanceType,auras) VALUES (@NPC,@path,0,0,1,0,0, '');
+INSERT INTO creature_addon (guid,path_id,mount,bytes1,bytes2,emote,visibilityDistanceType,auras) VALUES (@NPC+4,@path,0,0,1,0,0, '');
+INSERT INTO creature_addon (guid,path_id,mount,bytes1,bytes2,emote,visibilityDistanceType,auras) VALUES (@NPC+8,@path,0,0,1,0,0, '');
+DELETE FROM waypoint_data WHERE id=@path;
+INSERT INTO waypoint_data (id,point,position_x,position_y,position_z,orientation,delay,move_type,action,action_chance,wpguid) VALUES
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1655187726583284000.sql:57]:
```suggestion
DELETE FROM `creature_addon` WHERE `guid` IN (@NPC, @NPC+4, @NPC+8);
INSERT INTO `creature_addon` (`guid`,`path_id`,`mount`,`bytes1`,`bytes2`,`emote`,`visibilityDistanceType`,`auras`) VALUES
(@NPC,@path,0,0,1,0,0, ''),
(@NPC+4,@path,0,0,1,0,0, ''),
(@NPC+8,@path,0,0,1,0,0, '');

DELETE FROM `waypoint_data` WHERE `id`=@path;
INSERT INTO `waypoint_data` (`id`,`point`,`position_x`,`position_y`,`position_z`,`orientation`,`delay`,`move_type`,`action`,`action_chance`,`wpguid`) VALUES
```

**Review by Nefertumm — CHANGES_REQUESTED** [review]
Please, address what I pointed out, ty!

File: data/sql/updates/pending_db_world/rev_1655187726583284000.sql [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+(45814, 3237, 3236, 0, 1, 0, 0, 1, 1, 0, -3814.7678, -1845.9219, 93.61283, 0.949117600917816162, 900, 0, 0, 1, 0, 0, 0, 0, 0, '', 0),
+(45815, 3237, 3236, 0, 1, 0, 0, 1, 1, 0, -3798.8208, -1829.7017, 94.17238, 4.366050243377685546, 900, 0, 0, 1, 0, 0, 0, 0, 0, '', 0),
+(45816, 3235, 3236, 0, 1, 0, 0, 1, 1, 0, -3807.5981, -1865.3707, 92.56358, 1.319851279258728027, 900, 0, 0, 1, 0, 0, 0, 0, 0, '', 0),
+/* Combo 3 */
+(45817, 3237, 0, 0, 1, 0, 0, 1, 1, 0, -3787.7153, -1848.3755, 93.918015, 2.55345463752746582, 1800, 0, 0, 1, 0, 2, 0, 0, 0, '', 0), -- .go xyz -3787.7153 -1848.3755 93.918015 Leader 
+(45818, 5827, 0, 0, 1, 0, 0, 1, 1, 0, -3814.7678, -1845.9219, 93.61283, 0.949117600917816162, 1800, 0, 0, 1, 0, 0, 0, 0, 0, '', 0),
+(45819, 3235, 0, 0, 1, 0, 0, 1, 1, 0, -3798.8208, -1829.7017, 94.17238, 4.366050243377685546, 1800, 0, 0, 1, 0, 0, 0, 0, 0, '', 0),
+(45820, 3236, 0, 0, 1, 0, 0, 1, 1, 0, -3807.5981, -1865.3707, 92.56358, 1.319851279258728027, 1800, 0, 0, 1, 0, 0, 0, 0, 0, '', 0);
+
+DELETE FROM `pool_creature` WHERE guid IN (45809, 45810, 45811, 45812, 45813, 45814, 45815, 45816, 45817, 45818, 45819, 45820);
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1655187726583284000.sql:26]:
```suggestion
DELETE FROM `pool_creature` WHERE `guid` IN (45809, 45810, 45811, 45812, 45813, 45814, 45815, 45816, 45817, 45818, 45819, 45820);
```

Comment (Nefertumm) [reply data/sql/updates/pending_db_world/rev_1655187726583284000.sql:26]: Only this one left :D

**Comment by Maelthyrr** [comment]
Recheck IG, and everything is fine. I pass in to be merge, but I let you fix the review ;)

Thanks for you work ! <3

File: data/sql/updates/pending_db_world/rev_1655187726583284000.sql [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+
+DELETE FROM `pool_pool` WHERE `pool_id` IN (391, 392, 393);
+INSERT INTO `pool_pool` (`pool_id`, `mother_pool`, `chance`, `description`) VALUES 
+(391, 394, 55, 'Brontus Kodo Pack Combo 1 -- 4 Barrens Kodo Pack'),
+(392, 394, 30, 'Brontus Kodo Pack Combo 2 -- Mixed Kodo Pack'),
+(393, 394, 15, 'Brontus Kodo Pack Combo 3 -- Brontus Pack');
+
+-- Pathing for Kodos with Brontus
+SET @NPC := 45809;
+SET @path := @NPC * 10;
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1655187726583284000.sql:49]: Path should be on upper case, but you can leave it as it is for this one if you want.

**Review by Nefertumm — APPROVED** [review]

**Comment by acidmanifesto** [comment]
Thank you for the PR.

**Pull request merged**